### PR TITLE
Added tests for wasserstein distance with tensorflow

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Original PR : https://github.com/GUDHI/gudhi-devel/pull/377 (That I can close)
Note that I could not merge (to keep credentials) because of test split, but the copyright mention @MathieuCarriere . I am OK if @MathieuCarriere back merges this PR in https://github.com/GUDHI/gudhi-devel/pull/377

* Added tensorflow support for cmake.
* tensorflow was already in test-requirements.txt for CI.
* I splitted the test (cf. test_wasserstein_with_tensors.py maybe test_wasserstein_automatic_differentiation.py would be a better name ?)

In wait of https://github.com/GUDHI/gudhi-devel/pull/365 that could also use this PR to split tests.

### CODE REVIEW

File: src/python/CMakeLists.txt [hunk 1]
```diff
@@ -501,6 +504,11 @@ if(PYTHONINTERP_FOUND)
       endif()
       add_gudhi_py_test(test_wasserstein_barycenter)
     endif()
+    if(OT_FOUND)
+      if(TENSORFLOW_FOUND AND EAGERPY_FOUND)
+        add_gudhi_py_test(test_wasserstein_with_tensors)
```
Comment (mglisse) [inline src/python/CMakeLists.txt:509]: If the file is about tests with tensorflow, I would put "tensorflow" in the name. With this name, I thought it had both tensorflow and pytorch tests.

File: src/python/test/test_wasserstein_with_tensors.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    Modification(s):
+      - YYYY/MM Author: Description of the modification
+"""
+
+from gudhi.wasserstein import wasserstein_distance as pot
+import numpy as np
+
+def test_wasserstein_distance_grad_tensorflow():
+    import tensorflow as tf
```
Comment (mglisse) [inline src/python/test/test_wasserstein_with_tensors.py:15]: Might as well import early with gudhi and numpy, if the whole file requires tensorflow.

**Review by mglisse — COMMENTED** [review]
ok once the 2 comments are addressed.

File: src/python/CMakeLists.txt (hunk 1 above)
Comment (VincentRouvreau) [reply src/python/CMakeLists.txt:509]: Sorry about that but this was my first intention. What would you say if I move `test_wasserstein_distance_grad` from `test_wasserstein_distance.py` in `test_wasserstein_with_tensors.py` ? Like that, if you have only pot, at least wasserstein is tested. If you have tensorflow and torch, test_wasserstein_with_tensors is tested (P.S. : I don't know why eagerpy is required here...)

Comment (mglisse) [reply src/python/CMakeLists.txt:509]: ok

Comment (VincentRouvreau) [reply src/python/CMakeLists.txt:510]: So did I on f0beb329

File: src/python/test/test_wasserstein_with_tensors.py (hunk 2 above)
Comment (VincentRouvreau) [reply src/python/test/test_wasserstein_with_tensors.py]: imports are now done at global level

**Review by mglisse — APPROVED** [review]

**Review by MathieuCarriere — APPROVED** [review]

**Pull request merged**